### DISCUSSION TITLE
Multi-variable stratification with Autogluon - How to ensure it works correctly and doesn't overfit?

### DISCUSSION THREAD

**Discussion by apavlo89** [autogluon/autogluon#3273] [Q&A]
So I am trying to do multi-variable stratification (stratify by label but also by two other columns/variables) but I am not sure this will work as intended with autogluon and need some guidance for my binary classification problem.

```
dataset['strata'] = dataset['LABEL'].astype(str) + "_" + dataset['VAR1'].astype(str) + "_" + dataset['VAR2'].astype(str)

# Now use this new variable for stratification
X = dataset.drop(["LABEL", "strata"], axis=1)
X = pd.get_dummies(X)
y = dataset["LABEL"]
strata = dataset["strata"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=168, stratify=strata)


####################################################################################################################
#train-test databases###############################################################################################
####################################################################################################################


#training data
train_data = X_train.copy()  # Make a copy of X_train
train_data['WINNER'] = y_train  # Add the target variable to the new dataframe

#test_data
test_data = X_test.copy()
test_data['WINNER'] = y_test


######################################################################################################################
#AUTOGLUON with train-test split######################################################################################
######################################################################################################################
# Define the AutoGluon task and train the model
metric = 'balanced_accuracy' #precision_weighted or roc auc next

time_limit_s=3600*24 #then set it in predictor as well

predictor = TabularPredictor(label='WINNER',eval_metric=metric, path='autogluon_model_balancedaccuracy_multistratification_v1').fit(train_data=train_data, presets='best_quality', time_limit=time_limit_s)
predictions = predictor.predict(test_data)
predictor.evaluate(test_data, silent=True)

leaderboard = predictor.leaderboard(test_data, extra_info=True)
print(leaderboard)

#####################################################################################################################

```

Will this work okay or do I need to do something else? Any help on this would be super appreciated.
